sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1240,7 +1240,9 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       case SqlBaseParser.IN =>
         invertIfNotDefined(In(e, ctx.expression.asScala.map(expression)))
       case SqlBaseParser.LIKE =>
-        invertIfNotDefined(Like(e, expression(ctx.pattern)))
+        val escapeOpt = Option(ctx.escapeChar).map(_.getText)
+        val like = Like(e, expression(ctx.pattern), escapeOpt)
+        invertIfNotDefined(like)
```

Comment:
Needs to check if `escapeOpt` has a valid char for an escape ?
In PostgreSQL;
```
postgres=# select * from t where a like 'a%c' escape 'invalid';
ERROR:  invalid escape string
HINT:  Escape string must be empty or one character.
```

Reply:
A good idea. I will treat it in escapeLikeRegex.